urdfenvs/urdfCommon/urdfEnv.py

```diff
@@ -14,15 +13,34 @@ def __init__(self, msg, observation, observationSpace):
         super().__init__(msg + msgExt)
 
     def getWrongObservation(self, o, os):
-        msgExt = ": "
-        for key in o.keys():
-            if not os[key].contains(o[key]):
-                msgExt += "Error in " + key
-                for i, val in enumerate(o[key]):
-                    if val < os[key].low[i]:
-                        msgExt += f"[{i}]: {val} < {os[key].low[i]}"
-                    elif val > os[key].high[i]:
-                        msgExt += f"[{i}]: {val} > {os[key].high[i]}"
+        msgExt = ":\n"
+        msgExt += self.checkDict(o, os)
+        return msgExt
+
+    def checkDict(self, o_dict, os_dict, depth=1, tabbing=''):
+        msgExt = ''
+
+
+        for key in o_dict.keys():
+
+            # todo: warning if o_dict[key] or especially os_dict[key] is an Dict()
+            #  (and key is not in there which results in a error)
+
```

Comment:
Can you address that? Or should I do that?

Reply:
You have resolved it, thank you.